### PR TITLE
Dbever/pro#2782 import connections from DBvisualizer third party product 

### PR DESCRIPTION
**Pull request by serjiokov**
Import connections from DBvisualizer third-party product fixed

### CODE REVIEW

**Comment by serjiokov** [comment]
https://github.com/user-attachments/assets/e3e2a0b6-bf3b-4420-bf53-bf3ab3170638

**Review by E1izabeth — APPROVED** [review]

**Review by HocKu7 — APPROVED** [review]

**Comment by MashaKorax** [comment]
Configurations parse error when import drivers :  Cassandra, DB2 Datadirect, DB2 Iseries,  azure synapce dedicated, DB2 zos, Derby server, Derby embedded, 
iScreen Shoter - DBeaver Enterprise Edition - 240722110625

**Comment by MashaKorax** [comment]
NPE when import driver Databricks
```
java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null
	at java.base/java.util.regex.Matcher.getTextLength(Unknown Source)
	at java.base/java.util.regex.Matcher.reset(Unknown Source)
	at java.base/java.util.regex.Matcher.<init>(Unknown Source)
	at java.base/java.util.regex.Pattern.matcher(Unknown Source)
	at org.jkiss.utils.CommonUtils.replaceAll(CommonUtils.java:1023)
	at org.jkiss.dbeaver.model.DatabaseURL.getPattern(DatabaseURL.java:184)
	at org.jkiss.dbeaver.ext.import_config.wizards.ConfigImportWizard.parseUrlAsDriverSampleUrl(ConfigImportWizard.java:269)
	at org.jkiss.dbeaver.ext.import_config.wizards.ConfigImportWizard.adaptConnectionUrl(ConfigImportWizard.java:241)
	at org.jkiss.dbeaver.ext.import_config.wizards.ConfigImportWizard.importConnection(ConfigImportWizard.java:170)
	at org.jkiss.dbeaver.ext.import_config.wizards.ConfigImportWizard.performFinish(ConfigImportWizard.java:92)
	at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:837)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:474)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:265)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4660)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1622)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1645)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1630)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1392)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4427)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4003)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at org.eclipse.ui.internal.handlers.WizardHandler$Import.executeHandler(WizardHandler.java:153)
	at org.eclipse.ui.internal.handlers.WizardHandler.execute(WizardHandler.java:279)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:277)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:98)
	at jdk.internal.reflect.GeneratedMethodAccessor55.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:299)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:233)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:174)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:165)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:204)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:444)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:442)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:464)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4660)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1622)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1645)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1630)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1392)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4427)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4003)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
	at com.dbeaver.application.DBeaverApplicationAdvanced.start(DBeaverApplicationAdvanced.java:97)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
```

File: plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisAbstractConfigurationCreator.java [hunk 1]
```diff
@@ -62,4 +65,22 @@ public File getConfigurationAsset(File folder) {
...
+            .flatMap(provider -> provider.getEnabledDrivers().stream())
+            .filter(d -> d.getName().equals(name))
+            .findFirst();
+        if (descriptor.isEmpty()) {
+            descriptor = registry.getDataSourceProviders().stream()
+                .flatMap(provider -> provider.getEnabledDrivers().stream())
+                .filter(d -> d.getName().contains(name))
+                .findFirst();
+        }
+        if (descriptor.isEmpty()) {
```
Comment (HocKu7) [inline plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisAbstractConfigurationCreator.java]: `return descriptor.orElse(null)`

File: plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisSshServerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    private String sshImplementationType;
+
+    public String getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
```
Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisSshServerConfiguration.java:49]: Maybe making it as recotd will be more suitable for our Java version

File: plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -8,7 +8,8 @@ Bundle-Activator: org.jkiss.dbeaver.ext.import_config.Activator
 Require-Bundle: org.eclipse.ui,
  org.eclipse.core.runtime,
  org.jkiss.dbeaver.core,
- org.eclipse.core.resources
+ org.eclipse.core.resources,
+ org.jkiss.dbeaver.net.ssh.jsch
```
Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF:12]: Excessive import?

Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF:12]: Excessive import?

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF:12]: Fixed

File: plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisSshServerConfiguration.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisSshServerConfiguration.java:49]: The class is a kind of POJO encapsulation specific for DBvisualizer products, which, in general, might depend on a particular version of DBvisualizer.

File: plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisAbstractConfigurationCreator.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.import_config/src/org/jkiss/dbeaver/ext/import_config/dbvis/DbvisAbstractConfigurationCreator.java]: Fixed

File: plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.import_config/META-INF/MANIFEST.MF:12]: Dependency require to be here.

**Comment by MashaKorax** [comment]
verified

**Pull request merged**